collectors/plugins.d/pluginsd_parser.c

```diff
@@ -49,11 +49,16 @@ PARSER_RC pluginsd_chart_action(void *user, char *type, char *id, char *name, ch
 {
     RRDSET *st = NULL;
     RRDHOST *host = ((PARSER_USER_OBJECT *) user)->host;
+    uuid_t  *guid = &((PARSER_USER_OBJECT *)user)->guid;
 
-    st = rrdset_create(
-        host, type, id, name, family, context, title, units,
-        plugin, module, priority, update_every,
-        chart_type);
+    st = rrdset_create_custom(
+        host, type, id, name, family, context, title, units, plugin, module, priority, update_every, chart_type,
+        host->rrd_memory_mode, host->rrd_history_entries, 0,
+        uuid_is_null(*guid) ? NULL : guid);
```

Comment:
I know it's kind of implied by `callocz()` of the `PARSER_USER_OBJECT` but I think you should add an explicit initialization during parser startup such as `uuid_clear(user->guid)` in `streaming_parser()` definition to make it more obvious.

Reply:
Same would be needed for `pluginsd_process()`.

Reply:
I will add in streaming parser -- in pluginsd_process the GUID is not used

Reply:
Ah I see what you mean for the other keywords